app/src/main/java/com/nmakademija/nmaakademija/fragment/SettingsFragment.java

```diff
@@ -0,0 +1,41 @@
+package com.nmakademija.nmaakademija.fragment;
+
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.support.v7.preference.CheckBoxPreference;
+import android.support.v7.preference.Preference;
+import android.support.v7.preference.PreferenceFragmentCompat;
+
+import com.nmakademija.nmaakademija.R;
+import com.nmakademija.nmaakademija.utils.Preferences;
+
+public class SettingsFragment extends PreferenceFragmentCompat {
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.preferences, rootKey);
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
```

Comment:
If you use `PreferenceManager.getDefaultSharedPreferences(context)` you don't need any additional logic. `PreferenceFragmentCompat ` will handle everything for you e.g saving the preferences that user selected.